src/els_dt_references.erl

```diff
@@ -96,12 +97,31 @@ find_all() ->
   find_by(Pattern).
 
 %% @doc Find by id
--spec find_by_id(any()) -> {ok, [item()]} | {error, any()}.
-find_by_id(Id) ->
-  Pattern = #els_dt_references{id = Id, _ = '_'},
+-spec find_by_id(poi_kind(), any()) -> {ok, [item()]} | {error, any()}.
+find_by_id(Kind, Id) ->
+  InternalId = {kind_to_category(Kind), Id},
+  Pattern = #els_dt_references{id = InternalId, _ = '_'},
   find_by(Pattern).
 
 -spec find_by(tuple()) -> {ok, [item()]}.
 find_by(Pattern) ->
   {ok, Items} = els_db:match(Pattern),
   {ok, [to_item(Item) || Item <- Items]}.
+
+-spec kind_to_category(poi_kind()) -> function | type | macro | record.
```

Comment:
The idea of this function is to include in the guards all the supported POIs for which it is possible to find a reference.